apps/backend/src/libs/modules/encrypt/libs/types/encrypt.type.ts

```diff
@@ -0,0 +1,15 @@
+type EncryptResult = {
+	hash: string;
+	salt: string;
+};
+
+type Encrypt = {
+	compare(data: {
+		password: string;
+		passwordHash: string;
+		salt: string;
+	}): Promise<boolean>;
+	encrypt(password: string): Promise<EncryptResult>;
```

Comment:
```suggestion
	encrypt(password: string): ReturnType<Encrypt["encrypt"]>;
```
you the same way then

Reply:
@what1s1ove If you mean, make it in `BaseEncrypt` (not in `Encrypt `), I have already did it in this way

Reply:
do it in both places